### PR TITLE
Add Psalm

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

File: .github/workflows/static-analysis.yml [hunk 1]
```diff
@@ -37,3 +37,29 @@ jobs:
...
+        uses: "shivammathur/setup-php@v2"
+        with:
+          coverage: "none"
+          php-version: "${{ matrix.php-version }}"
+
+      - name: "Install dependencies with Composer"
+        uses: "ramsey/composer-install@v1"
+
+      - name: "Run a static analysis with vimeo/psalm"
+        run: "vendor/bin/psalm"
```
Comment (greg0ire) [inline .github/workflows/static-analysis.yml]: We usually use some extra options: https://github.com/doctrine/.github/blob/2c60b2f943cf08df21be3dc733f1b92db8d24beb/workflow-templates/static-analysis.yml#L63

Are those maybe not required anymore?

File: .github/workflows/static-analysis.yml [hunk 2]
```diff
@@ -37,3 +37,29 @@ jobs:
...
+        continue-on-error: true
```
Comment (greg0ire) [inline .github/workflows/static-analysis.yml]: Why would we want the job to pass if this step fails?

Comment (VincentLanglet) [reply .github/workflows/static-analysis.yml]: I don't know, this was like this for phpstan. So I supposed you didn't want it to be a required build
Fixed

File: .github/workflows/static-analysis.yml (hunk 1 above)
Comment (VincentLanglet) [reply .github/workflows/static-analysis.yml]: Done

**Review by greg0ire — APPROVED** [review]

**Comment by greg0ire** [comment]
I did not spot that phpstan was allowed to fail in https://github.com/doctrine/common/pull/910, maybe we could not find another way? An ignore rule would have been better. I'd fix it but I'm on my phone.

**Comment by VincentLanglet** [comment]
> I did not spot that phpstan was allowed to fail in #910, maybe we could not find another way? An ignore rule would have been better. I'd fix it but I'm on my phone.

Phpstan errors are:
```
 ------ --------------------------------------------------------------------
  Line   lib/Doctrine/Common/Proxy/ProxyGenerator.php
 ------ --------------------------------------------------------------------
  1119   Class ReflectionUnionType not found.
         💡 Learn more at https://phpstan.org/user-guide/discovering-symbols
  1124   Call to method getTypes() on an unknown class ReflectionUnionType.
         💡 Learn more at https://phpstan.org/user-guide/discovering-symbols
 ------ --------------------------------------------------------------------

 ------ --------------------------------------------------------------------
  Line   tests/Doctrine/Tests/Common/Proxy/ProxyMagicMethodsTest.php
 ------ --------------------------------------------------------------------
  417    Class Doctrine\Tests\Common\Proxy\MagicIssetClassWithInteger not
         found.
         💡 Learn more at https://phpstan.org/user-guide/discovering-symbols
 ------ --------------------------------------------------------------------
```

I think it's better to fix it in another PR

**Comment by greg0ire** [comment]
Me too FTR

**Review by SenseException — APPROVED** [review]
Yes, it can be handled in another PR.

**Comment by greg0ire** [comment]
Thanks @VincentLanglet !

**Comment by VincentLanglet** [comment]
Hi @greg0ire does a 3.1.2 release would be possible ? :)

**Comment by greg0ire** [comment]
Hi! Sure! https://github.com/doctrine/common/releases/tag/3.1.2

**Comment by VincentLanglet** [comment]
Thanks

**Pull request merged**